src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Options/PackageSourceMappingPage.cs

```diff
@@ -171,7 +168,7 @@ private static ExternalSettingOperationResult<T> GetValuePackageSourceMappings<T
                 {
                     string packageIdOrPattern = packageSourceMapping.Key;
                     List<PackageSourceContextInfo> packageSources = packageSourceMapping.Value;
-                    IEnumerable<string> packageSourceNames = packageSources.Select(source => source.Name);
+                    List<string> packageSourceNames = packageSources.Select(source => source.Name).ToList();
```

Comment:
nitpick:  from DevDiv's performance best practises document: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/23126/Performance-Best-Practices?anchor=minimizing-memory-allocation

> Supply capacity arguments to collection constructors (e.g. new List<int>(100) instead of new List<T>()) to avoid repeated allocations as the collection grows

While the ideal number of sources per mapping is just one, I don't think we have telemetry to help us verify that it's under 4 for all customers, or otherwise still small numbers.  But in any case, it's just changing one line into two, so something I think worthwhile always doing:

```suggestion
                    List<string> packageSourceNames = new(packageSources.Count);
                    packageSourceNames.AddRange(packageSources.Select(source => source.Name));
```